ban/core/resource.py

```diff
@@ -149,9 +141,10 @@ def coerce(cls, id, identifier=None):
             return cls.get(getattr(cls, identifier) == id)
         except cls.DoesNotExist:
             # Is it an old identifier?
-            from .versioning import IdentifierRedirect
-            new = IdentifierRedirect.follow(cls, identifier, id)
-            if new:
-                return cls.get(getattr(cls, identifier) == new)
-            else:
-                raise
+            from .versioning import Redirect
+            redirects = Redirect.follow(cls.__name__, identifier, id)
+            if redirects:
+                if len(redirects) > 1:
+                    raise MultipleRedirectsError(identifier, id, redirects)
+                raise RedirectError(identifier, id, redirects[0])
+            raise
```

Comment:
That's OK, I misread the fact that we are already within a `DoesNotExist` block.